packages/api/src/Base.ts

```diff
@@ -517,22 +530,29 @@ export default abstract class ApiBase<ApiType> {
     const storage = storageFromMeta(this.runtimeMetadata);
     const constants = constantsFromMeta(this.runtimeMetadata);
 
+    // only inject if we are not a clone (global init)
+    if (!this._options.source) {
+      Event.injectMetadata(this.runtimeMetadata);
+      Method.injectMethods(extrinsics);
+
+      // detect the extrinsic version in-use based on the last block
+      const lastBlock: SignedBlock = await this._rpcCore.chain.getBlock().toPromise();
```

Comment:
We now determine the extrinsic version from the chain - basically, based on the last block, we pull out the version from contained extrinsics and then use this in any future construction of this type.